core/src/main/scala/pickling/Pickler.scala

```diff
@@ -26,6 +27,7 @@ trait SPickler[T] {
 
 object SPickler {
   def generate[T]: SPickler[T] = macro Compat.PicklerMacros_impl[T]
+  implicit def spuToSp[T: SPicklerUnpickler]: SPickler[T] = implicitly[SPicklerUnpickler[T]].pickler
```

Comment:
I don't think it would change the macro, I imagined it like this:
```
trait SPicklerUnpickler[T] extends SPickler[T] with Unpickler[T] {
  def pickler: SPickler[T]
  override final def pickle(picklee: T, builder: PBuilder): Unit = 
    pickler.pickle(picklee, builder)
  override final def tag: FastTypeTag[T] = pickler.tag
}
```
Possibly it should be a final class and not a trait, even.

For collisions, I think you just don't want to have two picklers in scope.

I guess either way can work. Just wondering if we can think of any rationale to go one way or the other.